src/main/java/io/vertx/core/net/impl/pool/SimpleConnectionPool.java

```diff
@@ -242,7 +264,27 @@ public ConnectionPool<C> contextProvider(Function<ContextInternal, EventLoopCont
   }
 
   private void execute(Executor.Action<SimpleConnectionPool<C>> action) {
-    sync.submit(action);
+    if (resumeExecutor == null) {
+      sync.submit(action);
+    } else {
+      final Executor.Continuation continuation = sync.submitAndContinue(action);
+      if (continuation != null) {
+        // single threaded
+        resumeExecutor.execute(() -> resumeAndContinue(continuation));
```

Comment:
The comment here will be used to add counters (that can just use  `putOrdered/lazySet` given that's a single-threaded thing. @vietj 
I'm NOT happy to allocate a whole new lambda for this, because `Continuation` is captured :(
To save this we should NOT make any `Continuation` and just provide an API on the executor like

```
// return false implies calling a resume later
boolean executeAndContinue(...);

// throw IllegalStateException if not called after a previous returning false resume/executeAndContinue
boolean resume();
```
This means that `resumeAndContinue` on the pool doesn't need anymore the `continuation` parameter